src/TestLisExpr.chpl

```diff
@@ -62,19 +62,109 @@ module TestLisExpr
     }
 
+     // tasks per locale based on locale0
+    const numTasks = here.maxTaskPar;
+    const Tasks = {0..#numTasks}; // these need to be const for comms/performance reasons
+    
+    // calculate sub-domain for task
+    inline proc calcBlock(task: int, low: int, high: int) {
+        var totalsize = high - low + 1;
+        var div = totalsize / numTasks;
+        var rem = totalsize % numTasks;
+        var rlow: int;
+        var rhigh: int;
+        if (task < rem) {
+            rlow = task * (div+1) + low;
+            rhigh = rlow + div;
+        }
+        else {
+            rlow = task * div + rem + low;
+            rhigh = rlow + div - 1;
+        }
+        return {rlow .. rhigh};
+    }
+
+    /* eval with a specific environment for the test expression return true if no errors */
+    proc test_eval_coforall(const in prog: string): bool {
+        var f = true;
+        var N = 10;
+        var D = {0..#N};
+        var A: [D] int = D;
+        var B: [D] int;
+        
+        // this could have the advantage of not creating array temps like the rest of arkouda does
+        try {
+            coforall loc in Locales {
+                on loc {
+                    coforall task in Tasks {
+                        // get local domain's indices
+                        var lD = A.domain.localSubdomain();
+                        // calc task's indices from local domain's indices
+                        var tD = calcBlock(task, lD.low, lD.high);
+                        
+                        // parse and check the program
+                        var ast = parse(prog);
+                        // allocate the env for variables
+                        var env = new owned Env();
+                        
+                        for i in tD {
+                            // addEnrtry redefines values for already existing entries
+                            env.addEntry("elt",A[i]); // add a symbol called "elt" and value for a
+                            
+                            // this version does the eval the in the enviroment which creates the symbol "ans"
+                            //var ans = env.lookup("ans").toValue(int).v; // retrieve value for ans
+                            //b = ans;
+                            
+                            // this version just returns the GenValue from the eval call
+                            var ans = eval(ast,env);
+                            B[i] = ans.toValue(int).v; // put answer into b
+                        }
+                    }
+                }
+                
+                writeln(A);
+                writeln(B);
+            }
+        }
+        catch errors: TaskErrors {
+            for e in errors {
+                // why does compile complain about not having a try her and not in the serial case???!
+                try! writeln((e: unmanaged Error).message());
+            }
+            f = false;
+        }
+        catch {
```

Comment:
compiler also complained about not having a blanket `catch` block here, which I don't understand

Reply:
The reason that you need this is because `TaskErrors` is a subclass of error and there are still other errors that can be thrown in this context.

I was able to reduce to only needing a single catch by catching the `Error` and then checking if that error was of type `TaskErrors`:
```
        catch e: Error {
          if e.type == TaskErrors {
            for error in e {
                // why does compile complain about not having a try her and not in the serial case???!
                writeln((e: unmanaged Error).message());
            }
          } else {
            writeln(e.message());
          }
          f = false;
        }
```

Do you feel any better about this approach? 

Either way, like you pointed out, the documentation seems to be wrong here and I will open an issue for updating that.

Reply:
Upon some further investigation, it appears that you could also just switch the `try` here to be a `try!` and then you would no longer need to handle the regular error case, but then if an error of another type is thrown it will halt the program (but that should be extremely rare in this code).

So you could switch it to be something like this:
```
try! { // <-- this line was switched to be a try! from a try
  // code here...
} catch errors: TaskErrors {
  // handle errors...
}
```

Reply:
Opened up https://github.com/chapel-lang/chapel/issues/20289 with one of the `TaskErrors` bugs.